lib/util/Components.js

```diff
@@ -419,14 +419,81 @@ function componentRule(rule, context) {
       return utils.isReturningJSX(ASTNode, strict) || utils.isReturningNull(ASTNode);
     },
 
+    getComponentNameFromJSXElement(node) {
+      if (node.type !== 'JSXElement') {
+        return null;
+      }
+      if (node.openingElement && node.openingElement.name && node.openingElement.name.name) {
+        return node.openingElement.name.name;
+      }
+      return null;
+    },
+
+    /**
+     *
+     * @param {object} node
+     * Getting the first JSX element's name.
+     */
+    getNameOfWrappedComponent(node) {
+      if (node.length && node[0].body) {
+        const body = node[0].body;
+        let componentName;
+        if (body.type === 'JSXElement') {
+          componentName = this.getComponentNameFromJSXElement(body);
+        } else if (body.type === 'BlockStatement') {
+          const innerBody = body.body;
+          innerBody.forEach(item => {
+            if (item.type === 'ReturnStatement') {
+              componentName = this.getComponentNameFromJSXElement(item.argument);
+            }
+          });
+        }
+        return componentName;
+      }
+      return null;
+    },
+
+    /**
+     * Get the list of names of components created till now
+     */
+    getDetectedComponents() {
+      const list = components.list();
+      const componentList = [];
+      Object.keys(list).forEach(key => {
+        const val = list[key];
+        if (val.node.type === 'ClassDeclaration') {
+          componentList.push(val.node.id.name);
+        } else if (val.node.type === 'ArrowFunctionExpression') {
+          if (val.node.parent && val.node.parent.type === 'VariableDeclarator' && val.node.parent.id) {
+            componentList.push(val.node.parent.id.name);
+          }
+        }
+      });
+      return componentList;
+    },
+
+    /**
+     *
+     * @param {object} node
+     * It will check wheater memo/forwardRef is wrapping existing component or
+     * creating a new one.
+     */
+    isJustAWrapper(node) {
+      const childComponent = this.getNameOfWrappedComponent(node.arguments);
```

Comment:
Can you add a test for a fully anonymous SFC - ie, something for which `component.displayName || component.name` will be `undefined`?